### PR TITLE
planner: fallback to static plan for partition table if having no global stats in dynamic prune mode

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/37535

Problem Summary:

Currently we recommended users to analyze global stats for partition table before enabling dynamic prune mode. If we enabled dynamic prune mode without generating global stats, optimizer will think table stats as pesudo which generates bad plan.

### What is changed and how it works?

A proper way is to fallback to static plan if having no global stats in query.

### Check List

Tests 

- [x] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/integration_test.go [hunk 1]
```diff
@@ -7131,6 +7131,58 @@ func TestCastTimeAsDurationToTiFlash(t *testing.T) {
...
+	}
+	tk.MustQuery("explain select * from t").CheckAt([]int{0, 3, 4}, rows)
+
+	tk.MustExec("CREATE TABLE t2 (a int) PARTITION BY RANGE (a) (PARTITION p0 VALUES LESS THAN (6),PARTITION p1 VALUES LESS THAN (11));")
+	tk.MustExec("insert into t2 values (1),(2),(3),(4),(7),(8),(9),(10)")
+	tk.MustExec("analyze table t2")
+	tk.MustExec("set @@tidb_partition_prune_mode='dynamic'")
+
+	// use static plan in dynamic mode due to having not global stats
+	tk.MustQuery("explain select * from t").CheckAt([]int{0, 3, 4}, rows)
```
Comment (Reminiscent) [inline planner/core/integration_test.go:7159]: Why not explain format = 'brief`?

**Review by Reminiscent — APPROVED** [review]

File: planner/core/integration_test.go (hunk 1 above)
Comment (qw4990) [reply planner/core/integration_test.go:7159]: +1

**Review by qw4990 — APPROVED** [review]

**Pull request merged**